kodilla-hibernate/src/test/java/com/kodilla/hibernate/invoice/dao/InvoiceDaoTestSuite.java

```diff
@@ -24,34 +24,36 @@ public void testInvoiceDaoSave() {
         Product product1 = new Product("Pencil");
         Product product2 = new Product("Rubber");
         Product product3 = new Product("Scissors");
+
         Item item1 = new Item(new BigDecimal(1.20), 2, new BigDecimal(2.40));
-        product1.getItem().add(item1);
         Item item2 = new Item(new BigDecimal(1.00), 5, new BigDecimal(5));
-        product2.getItem().add(item2);
         Item item3 = new Item(new BigDecimal(5.20), 1, new BigDecimal(5.20));
-        product3.getItem().add(item3);
 
-        Invoice invoice = new Invoice("2/02/18");
+        item1.setProduct(product1);
+        item2.setProduct(product2);
+        item3.setProduct(product3);
+
+        Invoice invoice = new Invoice("5/02/18");
 
         invoice.getItems().add(item1);
         invoice.getItems().add(item2);
         invoice.getItems().add(item3);
 
-        item1.setInvoice(invoice);
-        item2.setInvoice(invoice);
-        item3.setInvoice(invoice);
-        item1.setProduct(product1);
-        item2.setProduct(product2);
-        item3.setProduct(product3);
-
         //when
-        invoiceDao.save(invoice);
-        int id = invoice.getId();
+        Invoice returnedInvoice = invoiceDao.save(invoice);
+        Product returnedProduct = returnedInvoice.getItems().get(0).getProduct();
+        System.out.println(returnedProduct);
+        int id = returnedInvoice.getId();
 
         //then
         Assert.assertNotEquals(0, id);
+        Assert.assertEquals(3, returnedInvoice.getItems().size());
 
         //cleanUp
-        // invoiceDao.delete(id);
+        invoiceDao.delete(id);
+
+        //after cleanUp
+        Assert.assertNotNull(product1);
```

Comment:
this does not make sense, product1 is always not null. 
You should check if products will be removed from DATABASE after removing invoice